src/main/java/org/example/App.java

```diff
@@ -0,0 +1,61 @@
+package org.example;
+
+
+import com.fasterxml.jackson.databind.json.JsonMapper;
+import model.Animal;
+import service.AnimalSerializer;
+
+import java.util.*;
+
+
+public class App {
+    public static void main(String[] args) {
+        Scanner scanner = new Scanner(System.in);
+        AnimalSerializer animalSerializer = new AnimalSerializer(new JsonMapper());
+        List<Animal> animals = AnimalSerializer.deserialize();
+        System.out.println(animals);
+        System.out.println("Do you want to add a pet? Yes / No");
+        String answer = scanner.next();
+
+
+        if (answer.equals("Yes")) {
+            System.out.println("Enter the name of the animal and owner name");
+            animals.add(new Animal(scanner.next(), scanner.next()));
+            animalSerializer.serialize(animals);
+            System.out.println(animals);
+        } else System.out.println("You don't add any pet");
+
+
+        System.out.println("Do you want to see all pets? Yes / No");
+        answer = scanner.next();
+
+
+        if (answer.equals("Yes")) {
+            System.out.println(AnimalSerializer.deserialize());
+        }
+
+        System.out.println("Do you want to took any pets? Yes / No");
+        answer = scanner.next();
+        if (answer.equals("Yes")) {
+            System.out.println("Enter the name of the pet you want to take");
+            String answerNameOfPet = scanner.next();
+            System.out.println(animals);
+
+            for (Animal animal : animals) {
```

Comment:
After fixingAnimalSerialyzer according to my comment you'll see that you are not able to modify this list in for cycle. Remove it and just create and serealize a new one
List<Animal> animals = animalSerializer.deserialize()
                        .stream().filter(animal -> !animal.getName().equals(answerNameOfPet)).collect(Collectors.toList());